bookkeeper-server/src/main/java/org/apache/bookkeeper/client/api/WriteHandle.java

```diff
@@ -41,7 +41,8 @@
     /**
      * Add entry asynchronously to an open ledger.
      *
-     * @param data array of bytes to be written
+     * @param data a bytebuf to be written. The bytebuf's reference count will be decremented by 1 after the
```

Comment:
I think it would be better to restate this, or to also state, that the append call passes ownership to of the bytebuf to the handle.

Reply:
I am not sure what do you expect to see about this. There is no ownership thing for this. The accurate statement is the bytebuf is decremented by 1 when the returned completablefuture is done. Because if the caller still has reference counts, the bytebuf is still 'owned' by the caller after the completable future is returned.

Reply:
In the normal case, the bytebuf will have a refcount of 1. It will be passed to append, and then the caller can forget about the bytebuf. In this case, the caller is passing ownership of that 1 refcount to the callee.
If the caller wants to keep a reference, it increments the refcount again. In this case, the callee will still own one refcount, and the caller will own another.

Not too fussed about this in any case. What's there is fine. Though the "completable future is returned" bit is unclear. Is it when the completable future completes, or when the append method returns?

Reply:
okay I will change it to "when the returned completablefuture is completed"